### ISSUE TITLE
CosmosDB provider support

### ISSUE THREAD

**Issue by whiskeyjay** [azdevops/terraform#37]

**Comment by whiskeyjay** [comment]
Tom is working on it:
https://github.com/terraform-providers/terraform-provider-azurerm/pull/108

**Comment by tombuildsstuff** [comment]
FYI [this is blocked on this Swagger/Generator issue](https://github.com/Azure/azure-rest-api-specs/issues/1324) - which means only one of the `consistencyLevel`'s currently works as expected right now

**Comment by marstr** [comment]
Heads up, after some digging, I found that this is being blocked by our autorest Go-Runtime, not the generator or swagger. For that reason, I created a new issue in that repository. You can find that issue here:
Azure/go-autorest#146

**Comment by tombuildsstuff** [comment]
@whiskeyjay this PR is now merged :)
